src/app/plan/components/Wizard/HooksStep.tsx

```diff
@@ -0,0 +1,220 @@
+        break;
+      }
+      case AddEditMode.Add: {
+        addHookRequest(hookValues);
+        setIsAddHooksOpen(false)
+        break;
+      }
+      default: {
+        console.warn(
+          `onAddEditSubmit, but unknown mode was found: ${hookAddEditStatus.mode}. Ignoring.`);
+      }
+    }
+  };
+
+  const columns = [
+    { title: 'Name' },
+    { title: 'Definition' },
+    { title: 'Run in' },
+    { title: 'Migration step' },
+  ];
+
+  let rows = []
+  let actions = [];
+  if (migHookList.length > 0) {
+    rows = migHookList.map((migHook, id) => {
+      return {
+        cells: [migHook.hookName, migHook.image, migHook.clusterTypeText, migHook.phase],
+      }
+    });
+    actions = [
+      {
+        title: 'Edit',
+        onClick: (event, rowId, rowData, extra) => {
+          const currentHook = migHookList.find((hook) => hook.hookName === rowData.name.title)
+          setInitialHookValues(currentHook)
+          setIsAddHooksOpen(true)
+          watchHookAddEditStatus(rowData.name.title);
+
+        }
+      },
+      {
+        title: 'Delete',
+        onClick: (event, rowId, rowData, extra) => {
+          removeHookRequest(rowData.name.title, rowData["migration-step"].title)
+        }
+      }
+    ]
+
+  }
+  if (rows.length === 0) {
+    rows = [{
+      heightAuto: true,
+      cells: [
+        {
+          props: { colSpan: 8 },
+          title: (
+            <Bullseye>
+              <EmptyState variant={EmptyStateVariant.full}>
+                <EmptyStateIcon icon={PlusCircleIcon} />
+                <Title headingLevel="h4" size="lg">
+                  No hooks have been added to this migration plan.
+                </Title>
+              </EmptyState>
+            </Bullseye >
+          )
+        },
+      ]
+    }]
+  }
+
+  const hooksFormContainerStyles = classNames(spacing.mtSm, styles.formContainerStyles)
+  return (
+    <Grid>
+      <GridItem className={spacing.mtSm}>
+        <TextContent>
+          <Text component={TextVariants.h2}>
+            Hooks
+          </Text>
+        </TextContent>
+      </GridItem>
+      <GridItem>
+        <TextContent>
+          <Text component={TextVariants.p}>
+            Hooks are commands that can be run at various steps in the migration process. They are defined in a container image or an Ansible playbook and can be run on either the source or target cluster.
+          </Text>
+        </TextContent>
+      </GridItem>
+      {isAddHooksOpen &&
+        <GridItem className={hooksFormContainerStyles}>
+          <HooksFormContainer
+            onAddEditHookSubmit={onAddEditHookSubmit}
+            initialHookValues={initialHookValues}
+            setInitialHookValues={setInitialHookValues}
+            setIsAddHooksOpen={setIsAddHooksOpen}
+            isAddHooksOpen={isAddHooksOpen}
+            isEditHook={false}
+            {...props} />
+        </GridItem>
+      }
+      {(!isAddHooksOpen) &&
+        <React.Fragment>
```

Comment:
I don't think you need a fragment here, you can just have the `isFetchingHookList ? <Bullseye> : <GridItem>` expression directly in the place of the fragment. You only need a fragment if you need more than one top-level node in your JSX expression.

Reply:
Needed it in this case for the nested conditional. There is probably another way to refactor this to make it cleaner.

Reply:
Interesting, I would have thought nesting it in parentheses would be enough. We can revisit it later maybe. Or not 😆